lib/moonshot/build_mechanism/script.rb

```diff
@@ -41,7 +41,7 @@ def build_hook(version)
 
   def post_build_hook(_version)
     unless File.exist?(@output_file) # rubocop:disable GuardClause
-      raise Thor::Error, 'Build command did not produce output file!'
+      step.failure 'Build command did not produce output file!'
```

Comment:
Not sure this will communicate the error upstream (like raise would) allowing a non-zero exit.